src/foam/u2/view/ReadReferenceView.js

```diff
@@ -9,97 +9,237 @@ foam.CLASS({
+      `
+    },
     {
       class: 'Boolean',
       name: 'enableLink',
-      documentation: 'Create the reference view as an anchor link to the reference\'s DetailView.',
+      documentation: `
+        Create the reference view as an anchor link to the reference's DetailView or provided menu.
+      `,
       value: true
+    },
+    {
+      class: 'StringArray',
+      name: 'menuKeys',
+      documentation: `
+        A list of menu ids.
+        The link will reference to the first menu to which group has permission
+        in this list. If no menus are permissioned, the link will be disabled.
+      `
     }
   ],
 
-  imports: [
-    'auth?',
-    'ctrl',
-    'stack'
-  ],
+  imports: ['auth?', 'ctrl', 'pushMenu', 'stack'],
 
   methods: [
     {
       name: 'initE',
       code: function() {
         var self = this;
         this.SUPER();
-        this
-          .add(this.obj$.map(obj => {
-            if ( ! obj ) return '';
-            if ( this.enableLink ) {
-              return this.E().start('a')
-                .attrs({href: '#'})
-                .on('click', function(evt) {
+        this.add(
+          this.obj$.map((obj) => {
+            if (!obj) return '';
+
+            if (this.enableLink) {
+              return this.E()
+                .start('a')
+                .attrs({ href: '#' })
+                .on('click', (evt) => {
                   evt.preventDefault();
-                  var pred = foam.mlang.predicate.False.create();
-                  self.stack.push({
-                    class:     'foam.comics.v2.DAOSummaryView',
-                    data:      self.obj,
-                    of:        self.of,
-                    backLabel: 'Back',
-                    config: self.DAOControllerConfig.create({
-                      daoKey: self.prop.targetDAOKey,
-                      createPredicate: pred,
-                      editPredicate: pred,
-                      deletePredicate: pred,
-                      editEnabled: false
-
-                    })
-                  }, self);
+                  if (self.linkTo === 'daoSummary') {
+                    const pred = foam.mlang.predicate.False.create();
+
+                    self.stack.push(
+                      {
+                        class: 'foam.comics.v2.DAOSummaryView',
+                        data: self.obj,
+                        of: self.obj.cls_,
+                        backLabel: 'Back',
+                        config: self.DAOControllerConfig.create({
+                          daoKey: self.prop.targetDAOKey,
+                          createPredicate: pred,
+                          editPredicate: pred,
+                          deletePredicate: pred,
+                          editEnabled: false
+                        })
+                      },
+                      self
+                    );
+                    // link to a menu
+                  } else {
+                    self.pushMenu(self.linkTo);
```

Comment:
Pushing to a menu - pushes to tableview - want Objects detail view

Reply:
could potentially - go with the logic similar to --     function calculateMemento(daoName) {
 from CapabilityGraphNodeView.js

Reply:
^ so that function - updates needed ... 
1) update the Memento handling in CapabilityGraphNodeView to update the memento with view. SEPERATOR and not an override the whole memento.
2) move the memento function from CapabilityGraphNodeView to a helper function in ApplicationController and export function into the context
3) import and use the helper function in CapabiltiyGraphNodeView and here in ReadReferenceView.js